### ISSUE TITLE
Bug: cannot override no-unused-vars

### ISSUE THREAD

**Issue by zayne-wang** [eslint/eslint#18194]
### Environment

Node version: v18.17.1
npm version: 10.4.0
Local ESLint version: 8.57.0
Global ESLint version: not installed, use local ESLint
Operating System: Windows 11 Professional 23H2

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```js
import js from "@eslint/js";

/**
 * @type {import("@types/eslint").Linter.FlatConfig}
 */
const config = {
  files: ["src/**/*.{js}"],
  rules: {
    "no-unused-vars": "off",
  },
};

export default [js.configs.recommended, config];
```
</details>

<details>
<summary>Tested code (src/test.js)</summary>

```js
const a = 1
```
</details>

### What did you expect to happen?

When execute `npx eslint --print-config ./eslint.config.js`, it should gives a feedback includes 

```js
"no-unused-vars": [
  0
]
``` 

### What actually happened?

Actually, the command gives a feedback keeps

```js
"no-unused-vars": [
  2
]
``` 

### Link to Minimal Reproducible Example

https://github.com/zayne-wang/eslint-cannot-override-rule

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

I'm not sure if it is a bug or only happens in my environment.

**Comment by Tanujkanti4441** [comment]
Hi @zayne-wang, thanks for the issue, but when i run the `npx eslint --print-config ./eslint.config.js` command in your example repo it gave me 
```js
"no-unused-vars": [
  0
]
```
here you can see
![Screenshot 2024-03-12 203157](https://github.com/eslint/eslint/assets/86398394/05980bf5-a2f5-4a43-ad0d-98b8ab52a708)

**Comment by zayne-wang** [comment]
> Hi @zayne-wang, thanks for the issue, but when i run the `npx eslint --print-config ./eslint.config.js` command in your example repo it gave me 
> 
> ```js
> 
> "no-unused-vars": [
> 
>   0
> 
> ]
> 
> ```
> 
> here you can see
> 
> ![Screenshot 2024-03-12 203157](https://github.com/eslint/eslint/assets/86398394/05980bf5-a2f5-4a43-ad0d-98b8ab52a708)
> 
> 

Hello @Tanujkanti4441 ! Thanks for your feedback, I will try to clear the cache tomorrow, and will give a feedback.

**Comment by Tanujkanti4441** [comment]
when i tested config is working with `files: ["src/**/*.js"]`.

**Comment by zayne-wang** [comment]
> when i tested config is working with `files: ["src/**/*.js"]`.

Thanks for your feedback. It's my fault to set wrong `files` property.
